src/ray/util/util.h

```diff
@@ -87,6 +89,9 @@ inline std::string AppendToEachLine(const std::string &str,
   return ss.str();
 }
 
+// Returns the TID of the calling thread.
+inline pid_t GetTid() { return syscall(__NR_gettid); }
```

Comment:
Does this work for mac osx and windows?

Reply:
Good catch! Just added some additional code for this. Will ping you once the Windows tests have run.